File: POO/Java/Polimorfismo.java
package POO.Java;

public class Polimorfismo {
    class Animal {
        void fazerSom() {
            System.out.println("Som genérico de animal.");
        }
    }
    
    class Gato extends Animal {
        @Override
        void fazerSom() {
            System.out.println("Miau!");
        }
    }
    
    class Vaca extends Animal {
        @Override
        void fazerSom() {
            System.out.println("Muuu!");
        }
    }
    
}

// Explicação: Mesmo que você chame fazerSom() 
// a partir de uma referência Animal, o som emitido 
// depende da classe real do objeto (Gato, Vaca, etc).

File: POO/Python/Polimorfismo.py
class Animal:
    def fazer_som(self):
        print("Som genérico de animal.")

class Gato(Animal):
    def fazer_som(self):
        print("Miau!")

class Vaca(Animal):
    def fazer_som(self):
        print("Muuu!")

# Explicação: Várias classes usam o mesmo método 
# (fazer_som), mas com comportamentos diferentes. 
# Isso é polimorfismo.

